tests/ci/cdk/pipeline/scripts/util.sh

```diff
@@ -99,8 +100,13 @@ function start_windows_img_build() {
   # Run commands on windows EC2 instance to build windows docker images.
   for i in {1..60}; do
     instance_id=$(aws ec2 describe-instances \
-      --filters "Name=tag:${WIN_EC2_TAG_KEY},Values=${WIN_EC2_TAG_VALUE}" | jq -r '.Reservations[0].Instances[0].InstanceId')
-    if [[ "${instance_id}" == "null" ]]; then
+      --filters "Name=tag:${WIN_EC2_TAG_KEY},Values=${WIN_EC2_TAG_VALUE}" \
+      --query 'Reservations[0].Instances[0].InstanceId' \
+      --output text 2>&1)
+    # if the build runs for too long, AWS sessions may expire and need to be refreshed
+    if echo "${instance_id}" | grep -q "RequestExpired"; then
+      refresh_session
+    elif [[ "${instance_id}" == "null" ]]; then
```

Comment:
Did you consider extending the duration in `assume_role` instead of using the short default time period?

Reply:
I tried but it's not possible to increase beyond the default time period because we're doing role chaining. According to this [documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html):

> Role chaining limits your AWS CLI or AWS API role session to a maximum of one hour. [...] However, if you assume a role using role chaining and provide a DurationSeconds parameter value greater than one hour, the operation fails.